app/src/main/java/net/gsantner/markor/frontend/MarkorDialogFactory.java

```diff
@@ -788,17 +788,32 @@ private static class Heading {
         }
     }
 
+    public static class HeadlineDialogState {
+        public Set<Integer> disabledLevels = new HashSet<>();
+        public String searchQuery = "";
+        public int listPosition = -1;
```

Comment:
@guanglinn Headline dialogs remember their position, query and the currently enabled levels